Lib/asyncio/queues.py

```diff
@@ -176,8 +204,13 @@ def get_nowait(self):
         """Remove and return an item from the queue.
 
         Return an item if one is immediately available, else raise QueueEmpty.
+
+        Raises QueueShutDown if the queue has been shut down and is empty, or
+        if the queue has been shut down immediately.
         """
         if self.empty():
```

Comment:
You have to check if queue is shutdown before testing the empty state.
This should be:
```
    if self._is_shutdown:
        raise QueueShutDown
    if self.empty():
        raise QueueEmpty
```

Reply:
No, while the queue still has items, `get()` should always return those items, regardless of the value of `_shutdown`. That is the whole point of `immediate=False`.

Reply:
> That is the whole point of immediate=False

You are right, inappropried remark, sorry !?